### PR TITLE
Support click behavior on the dynamic combo chart

### PR DESCRIPTION
**Pull request by alxnddr**
Closes https://github.com/metabase/metabase/issues/36293

### Description

Implements click behavior for the ECharts dynamic combo chart.

### How to verify

Verify separately on **1 dimension + N metrics** and **breakout charts: 2 dimensions 1 metric**

**Query Builder**
Click on series data and ensure all menu options work.  

**Dashboard**
Click on series data and ensure all menu options work.
Set up click different click behavior types and ensure they work propagating the right values.

### Demo

https://github.com/metabase/metabase/assets/14301985/71f3f77c-93dc-4da1-bfea-4219b1cef6bb

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by alxnddr** [comment]
Current dependencies on/for this PR:
* master
  * **PR #36489** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36489?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #36486** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36486?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #36135** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36135?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #36563** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36563?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #36317** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36317?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #36502** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36502?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #36510** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36510?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #36560** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36560?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #36548** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36548?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #36013** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36013?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35997** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35997?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35923** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35923?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35922** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35922?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35783** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35783?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #35996** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35996?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35573** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35573?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35395** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35395?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35345** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35345?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35321** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35321?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35307** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35307?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35304** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35304?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35303** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35303?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35301** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35301?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35300** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35300?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35297** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35297?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35118** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35118?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #34820** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34820?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #34923** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34923?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #34717** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34717?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #34772** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34772?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #34809** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34809?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #34816** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34816?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #34859** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34859?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #34869** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34869?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #34872** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34872?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #35068** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35068?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #34703** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34703?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #34763** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34763?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/36563?utm_source=stack-comment).

File: frontend/src/metabase/lib/objects.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const getObjectEntries = <K extends string, V>(
```
Comment (alxnddr) [inline frontend/src/metabase/lib/objects.ts:1]: This is only for preserving type information

File: frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts [hunk 2]
```diff
@@ -62,28 +74,35 @@ export const getDatasetKey = (
  * @param {DatasetColumn[]} columns - The columns of the raw dataset.
  * @param {RowValue[]} row - The raw row of values.
  * @param {number} cardId - The ID of the card.
- * @param {number} [breakoutIndex] - The breakout column index for charts with two dimension columns selected.
+ * @param {number} dimensionIndex - The dimension column index.
+ * @param {number} breakoutIndex - The breakout column index for charts with two dimension columns selected.
  */
-const aggregateMetricsForDatum = (
+const aggregateColumnValuesForDatum = (
```
Comment (alxnddr) [inline frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts:80]: This is for tooltips where we currently show non-metric values from the dataset that are not a dimension or a breakout

File: frontend/src/metabase/visualizations/echarts/cartesian/option/index.ts [hunk 3]
```diff
@@ -54,7 +54,6 @@ export const getCartesianChartOption = (
   ];
 
   return {
-    animation: false,
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/echarts/cartesian/option/index.ts:57]: Why remove this?

File: frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts [hunk 4]
```diff
@@ -99,6 +100,7 @@ const buildEChartsLineAreaSeries = (
     settings["stackable.stack_type"] != null ? `area_${yAxisIndex}` : undefined;
 
   return {
+    id: seriesModel.dataKey,
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts:103]: Will we need to add these to the scatter plot series as well?

File: frontend/src/metabase/visualizations/echarts/cartesian/option/index.ts (hunk 3 above)
Comment (alxnddr) [reply frontend/src/metabase/visualizations/echarts/cartesian/option/index.ts:57]: If we disable animation globally, hover color transitions are immediate and do not look good

**Review by EmmadUsmani — APPROVED** [review]

File: frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts (hunk 4 above)
Comment (alxnddr) [reply frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts:106]: Yep, it will be needed for hovers on scatter charts with breakouts

**Pull request merged**